storage/src/adb/sync/engine.rs

```diff
@@ -231,25 +231,25 @@ where
 
         for _ in 0..num_requests {
             // Convert fetched operations to operation counts for shared gap detection
-            let operation_counts: BTreeMap<u64, u64> = self
+            let operation_counts: BTreeMap<Location, u64> = self
                 .fetched_operations
                 .iter()
                 .map(|(&start_loc, operations)| (start_loc, operations.len() as u64))
                 .collect();
 
             // Find the next gap in the sync range that needs to be fetched.
             let Some((start_loc, end_loc)) = crate::adb::sync::gaps::find_next(
-                log_size,
-                self.target.upper_bound_ops,
+                Location::new(log_size),
+                self.target.upper_bound,
                 &operation_counts,
                 self.outstanding_requests.locations(),
-                self.fetch_batch_size.get(),
+                self.fetch_batch_size,
             ) else {
                 break; // No more gaps to fill
             };
 
             // Calculate batch size for this gap
-            let gap_size = NZU64!(end_loc - start_loc + 1);
+            let gap_size = NZU64!(end_loc.checked_sub(start_loc.as_u64()).unwrap().as_u64() + 1);
```

Comment:
nit: a lot of type conversion here. Wondering if there is a way to collapse this at all?